Gems/Atom/RHI/Vulkan/Code/Source/RHI/FrameGraphCompiler.h

```diff
@@ -100,6 +101,28 @@ namespace AZ
             VkPipelineStageFlags srcPipelineStageFlags = prevScopeAttachment ? GetResourcePipelineStateFlags(*prevScopeAttachment) : VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
             VkAccessFlags srcAccessFlags = prevScopeAttachment ? GetResourceAccessFlags(*prevScopeAttachment) : 0;
 
+            // this makes vkCmdClearColorImage is apart of this resource barrier
+            if constexpr (AZStd::is_same_v<ResourceScopeAttachment, RHI::ImageScopeAttachment>)
+            {
+                const auto& bindingDescriptor = scopeAttachment.GetDescriptor();
+                const auto& usageAndAccess = scopeAttachment.GetUsageAndAccess();
+                const bool isClearAction = bindingDescriptor.m_loadStoreAction.m_loadAction == RHI::AttachmentLoadAction::Clear;
+                const bool isClearActionStencil =
+                    bindingDescriptor.m_loadStoreAction.m_loadActionStencil == RHI::AttachmentLoadAction::Clear;
+
+                if ((isClearAction || isClearActionStencil) &&
+                    AZStd::any_of(
+                        usageAndAccess.begin(),
+                        usageAndAccess.end(),
+                        [](auto& usage)
+                        {
+                            return usage.m_usage == RHI::ScopeAttachmentUsage::Shader;
```

Comment:
Should this also be checking for ScopeAttachmentUsage::RenderTarget?

Reply:
CollectClearActions only applies to RHI::ScopeAttachmentUsage::Shader. feel like I should move this code to a common area since its just a copy out of Scope.cpp:322

Reply:
Ahh right. I forget why we dont allow clear operation on a rendertarget but it would have been nice to have a comment about it within CollectClearActions. This is fine other than moving the common logic into one place.

Reply:
I ended up moving it into a common place so the dependency is clear.

Reply:
yep. looks good.